nighthawk/source/common/rate_limiter_impl.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "common/common/logger.h"
+#include "envoy/event/timer.h"
+
+#include "nighthawk/common/rate_limiter.h"
+
+#include "frequency.h"
+
+namespace Nighthawk {
+
+// Simple rate limiter that will allow acquiring at a linear pace.
+// The average rate is computed over a timeframe that starts at
+// instantiation.
```

Comment:
Maybe comment on thread safety, as it's an open design question whether this should be shared or per-worker.